src/System.IO/src/System/IO/BufferedStream.cs

```diff
@@ -234,7 +236,11 @@ protected override void Dispose(bool disposing)
             finally
             {
                 _stream = null;
-                _buffer = null;
+                if (_buffer != null)
+                {
+                    ArrayPool<byte>.Shared.Return(_buffer);
```

Comment:
> What unexpected consequences are you referring to?

e.g. you load a .zip file using System.IO.Compression, and it contains some passwords.  Those passwords are now in the buffers.  Those buffers get returned to the pool, no one else uses the buffers from the pool, and they remain there for the remainder of the lifetime of the process.  Previously they would have been much more likely to be collected and overwritten on the heap, but now they remain fresh.